subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1264,6 +1267,12 @@ private int skipStackTrace(List<String> lines, int i) {
                 return i;
             }
 
+            // https://youtrack.jetbrains.com/issue/KT-29546
+            private boolean isKotlinCompilerFlakyStacktrace(String line, List<String> lines) {
+                return lines.stream().anyMatch(it -> it.contains("java.rmi.UnmarshalException: Error unmarshaling return header; nested exception is"))
```

Comment:
I think this check should only be done once, and not for each line.

Additionally to the `UnmarshalException` (or maybe instead of the `UnmarshalException`), we should probably match on `Compilation with Kotlin compile daemon was not successful`. WDYT?

Reply:
Note that the stacktrace contains two parts:

Part1:

```
Compilation with Kotlin compile daemon was not successful
java.rmi.UnmarshalException: Error unmarshaling return header; nested exception is: 
	java.io.EOFException
	at java.rmi/sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:254)
	at java.rmi/sun.rmi.server.UnicastRef.invoke(UnicastRef.java:164)
...
```

Part2:

```
Caused by: java.io.EOFException
	at java.base/java.io.DataInputStream.readByte(DataInputStream.java:271)
	at java.rmi/sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:240)
	... 116 more
```

So we have to catch these two parts in one method (because stacktrace matcher only matches `at XXXX`, not `Caused by:`) - I don't want to catch `Caused by: java.io.EOFException` in the 2nd part because that might skip real problems.